<div class='alert alert-warning'>

SciPy's interactive examples with Jupyterlite are experimental and may not always work as expected. Execution of cells containing imports may result in large downloads (up to 60MB of content for the first import from SciPy). Load times when importing from SciPy may take roughly 10-20 seconds. If you notice any problems, feel free to open an [issue](https://github.com/scipy/scipy/issues/new/choose).

</div>

In epidemiology, individuals are classified as "exposed" or
"unexposed" to some factor or treatment. If the occurrence of some
illness is under study, those who have the illness are often
classified as "cases", and those without it are "noncases".  The
counts of the occurrences of these classes gives a contingency
table
```

exposed    unexposed
cases          a           b
noncases       c           d

```
The sample odds ratio may be written ``(a/c) / (b/d)``.  ``a/c`` can
be interpreted as the odds of a case occurring in the exposed group,
and ``b/d`` as the odds of a case occurring in the unexposed group.
The sample odds ratio is the ratio of these odds.  If the odds ratio
is greater than 1, it suggests that there is a positive association
between being exposed and being a case.

Interchanging the rows or columns of the contingency table inverts
the odds ratio, so it is important to understand the meaning of labels
given to the rows and columns of the table when interpreting the
odds ratio.

In [4], the use of aspirin to prevent cardiovascular events in women
and men was investigated. The study notably concluded:

    ...aspirin therapy reduced the risk of a composite of
    cardiovascular events due to its effect on reducing the risk of
    ischemic stroke in women [...]

The article lists studies of various cardiovascular events. Let's
focus on the ischemic stoke in women.

The following table summarizes the results of the experiment in which
participants took aspirin or a placebo on a regular basis for several
years. Cases of ischemic stroke were recorded
```

Aspirin   Control/Placebo
Ischemic stroke     176           230
No stroke         21035         21018

```
The question we ask is "Is there evidence that the aspirin reduces the
risk of ischemic stroke?"

Compute the odds ratio:


In [ ]:
from scipy.stats.contingency import odds_ratio
res = odds_ratio([[176, 230], [21035, 21018]])
res.statistic

0.7646037659999126

For this sample, the odds of getting an ischemic stroke for those who have
been taking aspirin are 0.76 times that of those
who have received the placebo.

To make statistical inferences about the population under study,
we can compute the 95% confidence interval for the odds ratio:


In [ ]:
res.confidence_interval(confidence_level=0.95)

ConfidenceInterval(low=0.6241234078749812, high=0.9354102892100372)

The 95% confidence interval for the conditional odds ratio is
approximately (0.62, 0.94).

The fact that the entire 95% confidence interval falls below 1 supports
the authors' conclusion that the aspirin was associated with a
statistically significant reduction in ischemic stroke.